iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/TopologyModificationUtils.java

```diff
@@ -464,13 +465,52 @@ private static void addOrderPositions(Connectable<?> connectable, VoltageLevel v
         }
     }
 
+    /**
+     * Utility method to get all the feeders on a voltage level by connectable.
+     */
+    public static Map<String, List<ConnectablePosition.Feeder>> getFeedersByConnectable(VoltageLevel voltageLevel) {
+        Map<String, List<ConnectablePosition.Feeder>> feedersByConnectable = new HashMap<>();
+        voltageLevel.getConnectables().forEach(connectable -> {
+            ConnectablePosition<?> position = (ConnectablePosition<?>) connectable.getExtension(ConnectablePosition.class);
+            if (position != null) {
+                List<ConnectablePosition.Feeder> feeder = getFeeders(position, voltageLevel, connectable, false, Reporter.NO_OP);
+                feedersByConnectable.put(connectable.getId(), feeder);
+            }
+        });
+        return feedersByConnectable;
+    }
+
     private static List<Integer> getOrderPositions(ConnectablePosition<?> position, VoltageLevel voltageLevel, Connectable<?> connectable, boolean throwException, Reporter reporter) {
+        List<ConnectablePosition.Feeder> feeders = new ArrayList<>();
+        if (connectable instanceof Injection) {
+            feeders = getInjectionFeeder(position);
+        } else if (connectable instanceof Branch) {
+            feeders = getBranchFeeders(position, voltageLevel, (Branch<?>) connectable);
+        } else if (connectable instanceof ThreeWindingsTransformer) {
+            feeders = get3wtFeeders(position, voltageLevel, (ThreeWindingsTransformer) connectable);
+        } else {
+            LOGGER.error("Given connectable not supported: {}", connectable.getClass().getName());
+            connectableNotSupported(reporter, connectable);
+            if (throwException) {
+                throw new AssertionError("Given connectable not supported: " + connectable.getClass().getName());
+            }
+        }
+        List<Integer> orders = new ArrayList<>();
+        feeders.forEach(feeder -> feeder.getOrder().ifPresent(orders::add));
+        checkConnectableInVoltageLevel(orders, voltageLevel, connectable, throwException, reporter);
```

Comment:
In this method we don't really check that the connectable is in voltage level, in fact now that we moved it we also get this error if the connectable is not supported (well so far that's not possible but still). I'd rather check earlier if connectable is in voltage level by just doing `connectable.getTerminals().stream().anyMatch(t -> t.getVoltageLevel() == voltageLevel)`, instead of supposing we can see this in orders.

Reply:
I agree, if you want to check if orders are empty, rename this method `checkOrdersExist`.
Before that, you can check that the connectable is in the voltage level as florian proposed.